### PR TITLE
Feature/lazy loading messages

### PR DESCRIPTION
**Pull request by siepra**

### CODE REVIEW

File: src/renderer/components/pages/Channel.tsx [hunk 1]
```diff
@@ -28,7 +28,11 @@ export interface ChannelComponentProps {
   channel: PublicChannel
   channelSettingsModal: ReturnType<typeof useModal>
   channelInfoModal: ReturnType<typeof useModal>
-  messages: { [date: string]: DisplayableMessage[][] }
+  messages: {
+    count: number
+    groups: { [date: string]: DisplayableMessage[][] }
```
Comment (kowalski) [inline src/renderer/components/pages/Channel.tsx:33]: I'm not 100% sure if storing groups in an object enumarted by date will not mess up the relative order of groups.
I know `Map` maintains order of insertions, but I'm really not sure about objects.
Maybe you should use sth like: 
```
[ 
[string, DisplayableMessage[][] ]
]
```
instead ?

**Review by kowalski — APPROVED** [review]

**Review by vinkabuki — APPROVED** [review]

File: src/renderer/components/pages/Channel.tsx (hunk 1 above)
Comment (siepra) [reply src/renderer/components/pages/Channel.tsx:33]: I've got tests for selectors and they seem fine. Also we've been chatting with Bartek (and even Holmes) a few times and it was working well every time

**Review by holmesworcester — CHANGES_REQUESTED** [review]
Are there tests to ensure lazy loading is working correctly on scroll up, down, etc? And that message ordering is correct and messages aren't missed?

If I'm missing them can you talk me through it via voice or on Slack?

**Comment by siepra** [comment]
It's impossible to test scrolling (or is it? @kowalski). We have tests for slicing messages and ordering them though

**Comment by kowalski** [comment]
> It's impossible to test scrolling (or is it? @kowalski). We have tests for slicing messages and ordering them though

Yes, in these tests we're using jsdom, so there is no actual screen, sizes, layout, etc. 
However we could test this particular thing in testcafe

**Comment by siepra** [comment]
We had this problem recently that we don't know how to access message's unique id form testcafe. We agreed that it's better to wait until frontend gets stable before we make more advanced e2e tests. Somewhere in the future we're probably gonna introduce highlighting separate messages (in order to display its creation date and make it possible to edit/remove them etc.) so maybe it'd be better to do it then

**Review by holmesworcester — APPROVED** [review]

**Pull request merged**